lib/spack/spack/test/concretize_preferences.py

```diff
@@ -174,3 +175,69 @@ def test_external_mpi(self):
         spec = Spec('mpi')
         spec.concretize()
         assert spec['mpich'].external_path == '/dummy/path'
+
+    def test_config_permissions(self):
```

Comment:
Can this be broken out into multiple tests, so it's easier to see when `all` fails vs. a specific package?